components/PlaybackTask.brs

```diff
@@ -0,0 +1,253 @@
+Library "Roku_Ads.brs" ' Must be initialized in the task (or main) thread
+
+sub init()
+  ? "TRUE[X] >>> PlaybackTask::init()"
+  m.top.functionName = "setup"
+end sub
+
+sub setup() 
+  setupScopedVariables()
+  setupVideo()
+  setupRaf()
+  setupEvents()
+  initPlayback()
+end sub
+
+sub setupScopedVariables()
+  m.port = createObject("roMessagePort")
+  m.adFacade = m.top.adFacade
```

Comment:
This is more for rendering the TrueX ad.  Similar to the player, that is passed up from the render thread to the task thread.  This was inspired from the Crackle sample and is similar to the existing ref app appending it to the render thread directly.
    ' instantiate TruexAdRenderer and register for event updates
    m.adRenderer = m.top.createChild("TruexLibrary:TruexAdRenderer") `

Reply:
OK I guess it's fine if you believe it's a best practice and a good way to do it. I just thought it was odd to use a `<Rectangle>` instead of a `<Group>`, is this to mask the underlying UI / player as we spin up the interactive experience?

Reply:
Group works fine (will changed to this).  I didn't concern myself too much with what the container was.

This is more to give something to attach the AdRenderer that comes from the render thread.  The player is handled by visibility state.